eventually-v0-5-0/src/event.rs

```diff
@@ -0,0 +1,111 @@
+//! Module `event` contains types and abstractions helpful for working
+//! with Domain Events.
+
+use std::fmt::Debug;
+
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use serde::{Deserialize, Serialize};
+
+use crate::{
+    version::{ToConflictError, Version},
+    Message, Messages,
+};
+
+/// An Event is a [Message] carring the information about a Domain Event,
+/// an occurrence in the system lifetime that is relevant for the Domain
+/// that is being implemented.
+pub type Event<T> = Message<T>;
+
+/// Shortcut type to represent multiple [Event]s.
+pub type Events<T> = Messages<T>;
```

Comment:
once again, i think this indirection is more confusing than helpful

Reply:
Why is that? This is what the linter suggests me to do due to some complex return types or bounds 👀

Reply:
which return type is clippy complaining about?

```rust
pub type Event<T> = Message<T>;
```

this alias doesn't reduce the complexity. It just means a consumer has to learn two names for the same object

Reply:
That alias specifically (`Event<T>`) is not about complexity, it's just representing the `Event` type as a `Message` (as intended).

Reply:
>which return type is clippy complaining about?

The one about `event::Store::stream`, due to the very complex type returned.

I guess you're right though about the `Events<T>` and `Messages<T>` shortcut. Will remove them 👍🏻